core/store/src/main/java/uk/gov/gchq/gaffer/store/schema/Schema.java

```diff
@@ -302,109 +200,230 @@ public String toString() {
-            getElementDefs().setVisibilityProperty(propertyName);
-            return this;
+        @JsonIgnore
+        public CHILD_CLASS json(final Path... filePaths) throws SchemaException {
+            return json(Schema.class, filePaths);
         }
 
-        public Builder timestampProperty(final String propertyName) {
-            getElementDefs().setTimestampProperty(propertyName);
-            return this;
+        @JsonIgnore
+        public CHILD_CLASS json(final byte[]... jsonBytes) throws SchemaException {
+            return json(Schema.class, jsonBytes);
         }
 
         @Override
         public Schema build() {
-            final Schema schema = (Schema) super.build();
-            if (!schema.validate()) {
-                throw new SchemaException("The schema is not valid. Check the logs for more information.");
+            for (final SchemaElementDefinition elementDef : getThisSchema().getEntities().values()) {
+                elementDef.schemaReference = getThisSchema();
             }
 
-            return schema;
+            for (final SchemaElementDefinition elementDef : getThisSchema().getEdges().values()) {
+                elementDef.schemaReference = getThisSchema();
+            }
+
+            expandElementDefinitions(getThisSchema());
+
+            getThisSchema().types = Collections.unmodifiableMap(getThisSchema().types);
+
+            return super.build();
         }
 
-        @Override
-        public Schema buildModule() {
-            return (Schema) super.buildModule();
+        private Schema getThisSchema() {
+            return getElementDefs();
+        }
+
+        private void validateSharedGroups(final Set<String> groupsA, final Set<String> groupsB) {
+            final Set<String> sharedGroups = new HashSet<>(groupsA);
+            sharedGroups.retainAll(groupsB);
+            if (!sharedGroups.isEmpty()) {
+                throw new SchemaException("Element groups cannot be shared across different schema files/parts. Each group must be fully defined in a single schema. Please fix these groups: " + sharedGroups);
+            }
+        }
+
+        private void expandElementDefinitions(final Schema schema) {
+            for (final Entry<String, SchemaEdgeDefinition> entry : Lists.newArrayList(schema.getEdges().entrySet())) {
+                schema.getEdges().put(entry.getKey(), entry.getValue().getExpandedDefinition());
+            }
+
+            for (final Entry<String, SchemaEntityDefinition> entry : Lists.newArrayList(schema.getEntities().entrySet())) {
+                schema.getEntities().put(entry.getKey(), entry.getValue().getExpandedDefinition());
+            }
+        }
+    }
+
+    @JsonPOJOBuilder(buildMethodName = "build", withPrefix = "")
+    public static final class Builder extends BaseBuilder<Builder> {
+        public Builder() {
+        }
+
+        public Builder(final Schema schema) {
+            this();
```

Comment:
I think it is useful - it is just a shortcut for new Builder().merge(schema)
Allows users to start from a existing schema and make change.